sdk/appconfiguration/app-configuration/types/app-configuration.d.ts

```diff
@@ -0,0 +1,387 @@
+     */
+    deleteConfigurationSetting(id: ConfigurationSettingId, options?: DeleteConfigurationSettingOptions): Promise<DeleteConfigurationSettingResponse>;
+    /**
+     * Gets a setting from the Azure App Configuration service.
+     *
+     * Example code:
+     * ```ts
+     * const setting = await client.getConfigurationSetting({ key: "MyKey", label: "MyLabel" });
+     * ```
+     * @param id The id of the configuration setting to get.
+     * @param options Optional parameters for the request.
+     */
+    getConfigurationSetting(id: ConfigurationSettingId, options?: GetConfigurationSettingOptions): Promise<GetConfigurationSettingResponse>;
+    /**
+     * Lists settings from the Azure App Configuration service, optionally
+     * filtered by key names, labels and accept datetime.
+     *
+     * Example code:
+     * ```ts
+     * const allSettingsWithLabel = await client.listConfigurationSettings({ labels: [ "MyLabel" ] });
+     * ```
+     * @param options Optional parameters for the request.
+     */
+    listConfigurationSettings(options?: ListConfigurationSettingsOptions): PagedAsyncIterableIterator<ConfigurationSetting, ListConfigurationSettingPage>;
+    private getListConfigurationSettingsIterator;
+    private listConfigurationSettingsByPage;
+    private createListConfigurationPageFromResponse;
+    /**
+     * Lists revisions of a set of keys, optionally filtered by key names,
+     * labels and accept datetime.
+     *
+     * Example code:
+     * ```ts
+     * const revisionsIterator = await client.listRevisions({ keys: ["MyKey"] });
+     * ```
+     * @param options Optional parameters for the request.
+     */
+    listRevisions(options?: ListRevisionsOptions): PagedAsyncIterableIterator<ConfigurationSetting, ListRevisionsPage>;
+    private getListRevisionsIterator;
+    private listRevisionsByPage;
+    /**
+     * Sets the value of a key in the Azure App Configuration service, allowing for an optional etag.
+     * @param key The name of the key.
+     * @param configurationSetting A configuration value.
+     * @param options Optional parameters for the request.
+     *
+     * Example code:
+     * ```ts
+     * await client.setConfigurationSetting({ key: "MyKey", value: "MyValue" });
+     * ```
+     */
+    setConfigurationSetting(configurationSetting: SetConfigurationSettingParam, options?: SetConfigurationSettingOptions): Promise<SetConfigurationSettingResponse>;
+    /**
+     * Sets a key's value to read only
+     * @param id The id of the configuration setting to set to read-only.
+     */
+    setReadOnly(id: ConfigurationSettingId, options?: SetReadOnlyOptions): Promise<SetReadOnlyResponse>;
+    /**
+     * Makes the key's value writable again
+     * @param id The id of the configuration setting to make writable.
+     */
+    clearReadOnly(id: ConfigurationSettingId, options?: ClearReadOnlyOptions): Promise<ClearReadOnlyResponse>;
```

Comment:
It's a little strange to me to have `setReadOnly` and `clearReadOnly` instead of `setReadOnly(true)`/`setReadOnly(false)`, but maybe that's just me picking at the underlying API

Reply:
User study will determine - this came up during the .net review as well.